dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/AsyncTaskExecutionContext.java

```diff
@@ -59,19 +59,17 @@ public AsyncTaskExecutionContext(@NonNull TaskExecutionContext taskExecutionCont
         this.executeInterval = Math.max(asyncTaskExecuteFunction.getAsyncTaskStateCheckInterval().toMillis(), 1000L);
     }
 
-    public void refreshStartTime() {
-        if (executeTimes == 0) {
+    public synchronized void refreshStartTime() {
+        if (executeTimes != 0) {
             // The first time doesn't have delay
-            executeTimes++;
-        } else {
             currentStartTime = System.currentTimeMillis();
         }
+        executeTimes++;
     }
 
     @Override
     public long getDelay(TimeUnit unit) {
-        long nextExecuteTimeDelay = Math.min(currentStartTime + executeInterval, timeout) - System.currentTimeMillis();
-        return unit.convert(nextExecuteTimeDelay, TimeUnit.MILLISECONDS);
+        return unit.convert(this.executeInterval, TimeUnit.MILLISECONDS);
```

Comment:
Have you test this? Once the element changed in the delay queue, the delay time will recalculate, in this pr the delay time will be a fixed value, this will cause some function will never be triggered.

Reply:
sorry for that, I changed it.